src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.Fixer.cs

```diff
@@ -0,0 +1,182 @@
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    title,
+                    async c => await ExtractConstArrayAsync(node, model, editor, generator, context.Diagnostics.First().Properties, c).ConfigureAwait(false),
+                    equivalenceKey: title),
+                context.Diagnostics);
+        }
+
+        private static async Task<Document> ExtractConstArrayAsync(SyntaxNode node, SemanticModel model, DocumentEditor editor,
+            SyntaxGenerator generator, ImmutableDictionary<string, string> properties, CancellationToken cancellationToken)
+        {
+            IArrayCreationOperation arrayArgument = GetArrayCreationOperation(node, model, cancellationToken, out bool isInvoked);
+            IEnumerable<ISymbol> typeFields = model.GetEnclosingSymbol(node.SpanStart, cancellationToken).ContainingType
+                .GetMembers().Where(x => x is IFieldSymbol);
+
+            // Get a valid member name for the extracted constant
+            string newMemberName = GetExtractedMemberName(typeFields.Select(x => x.Name), properties["paramName"] ?? GetMemberNameFromType(arrayArgument));
+
+            // Get method containing the symbol that is being diagnosed. Should always be in a method
+            IOperation? containingMethodBody = arrayArgument.GetAncestor<IMethodBodyOperation>(OperationKind.MethodBody);
+            containingMethodBody ??= arrayArgument.GetAncestor<IBlockOperation>(OperationKind.Block); // VB methods have a different structure than CS methods
+
+            RoslynDebug.Assert(containingMethodBody != null);
+
+            // Create the new member
+            SyntaxNode newMember = generator.FieldDeclaration(
+                newMemberName,
+                generator.TypeExpression(arrayArgument.Type),
+                GetAccessibility(model.GetEnclosingSymbol(containingMethodBody!.Syntax.SpanStart, cancellationToken)),
+                DeclarationModifiers.Static | DeclarationModifiers.ReadOnly,
+                arrayArgument.Syntax.WithoutTrailingTrivia() // don't include extra trivia before the end of the declaration
+            ).FormatForExtraction(containingMethodBody.Syntax); // any additional formatting
+
+            // Add the new extracted member
+            ISymbol lastFieldSymbol = typeFields.LastOrDefault();
+            if (lastFieldSymbol is not null)
+            {
+                // Insert after last field if any fields are present
+                SyntaxNode lastFieldNode = await lastFieldSymbol.DeclaringSyntaxReferences.First().GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+                editor.InsertAfter(generator.GetDeclaration(lastFieldNode), newMember);
+            }
+            else
+            {
+                // Insert before first method if no fields are present, as a method already exists
+                editor.InsertBefore(containingMethodBody.Syntax, newMember);
```

Comment:
How is this "first method"?

Reply:
Hmmm, I have no idea what I was thinking. That feature is still important, so I'll have to get all method symbols, unless there is a better way to do it that you can think of.

Reply:
@steveberdy actually we're looking for methods here? Why not put it at the top of the class? ie, before the first class member?

Reply:
Well, members can refer to fields, properties, and methods. I think there is a way to get all members, but I'll have to check. My logic here, as I remember, was that by that block there would be no class fields, and therefore I'd have to look for some other type of member. It isn't perfect logic looking back, so I'll jump in and see what I can do.

Reply:
Looks like I fixed that a while back so I'm resolving. Thanks